ci: enable test-appetize for deployment/* branches

jfmcquade opened the pull request:
PR Checklist

- [x] - Latest `master` branch merged
- [x] - PR title descriptive (can be used in release notes)

## Description

Allows PRs made to "deployment" branches (e.g. `deployment/wash`) to trigger the "Test Appetize" github action and thereby deploy a preview Android build to appetize.io. This is achieved in the same way as it is when making a PR to the master branch, namely by adding the `test - appetize` label to the PR on github.

I've set up a new appetize account on IDEMS's behalf. Each app deployment has an associated app on appetize.io. I've invited @esmeetewinkel to the account, but not @chrismclarke as appetize says of invitations, "All of the user's existing and future apps will be added to the account", which is not desirable.

## Testing

The `test - appetize` label is on this PR, and the action is running correctly in the case of a PR to the master branch. After merging, we can test the functionality for a PR to a `deployment/*` branch.

## Dev notes

In order to handle the different deployments deploying to different appetize endpoints, I have set up [github environments](https://github.com/IDEMSInternational/parenting-app-ui/settings/environments) to store the respective secrets for each deployment. These environments relating to the app deployments could have further uses in the future – the pattern for dynamically setting the github actions environment based upon the branch name is contained in [test-appetize.yml](https://github.com/IDEMSInternational/parenting-app-ui/pull/1864/files#diff-125f09497de8e01f26e07df779ef6fcb6aa80a615a5c08d69b27313171268e6b). However these environments will likely be usurped once we have separate content repos.
NB: each APPETIZE_TOKEN secret is identical, and refers to the API token for the appetize account. I didn't want to change the general repo secret until this branch was pushed and tested, but once it is we can overwrite that repo-level secret with the new token and delete the secrets that exist on the environment level.

## Git Issues

Closes #1829

chrismclarke reviewed (changes requested):
Sounds useful to me, Appetize can support multiple apps so hopefully it should just automatically detect the new deployment from the updated app id (if not perhaps will need to populate first upload manually)

Whilst working on this could you please also set up an appetize account on behalf of IDEMS? The api token can then be populated to github secrets

We're still using my personal account and starting to get close to my 100 min per month free allowance

chrismclarke commented on .github/workflows/test-appetize.yml:
```diff
@@ -5,15 +5,45 @@ on:
+      # https://www.linuxjournal.com/article/8919, https://github.com/greenbone/gsa/actions/runs/1277581365/workflow
+      - name: Set Environment Deployment
+        run: |
+          if [ "$GITHUB_BASE_REF" = "master" ];                    \
```
nit(blocking) 
Do we need to be repeating this? I thought it was already calculated during the `set_environment` job?

chrismclarke reviewed (changes requested):
Really nice to see the separation of setting environment vs main script (and ability to override env vars) - super powerful pattern to have working.

I appreciate testing all this is a total pain, but can you check the question I raised about repeating environment setting?

jfmcquade commented:
> Haha, not a proper CI PR without at least 20 commits all just testing to see if things finally work 😅

I've added a few more for good measure...

You raise a good point about repeating the DEPLOYMENT_NAME calculation. I've refactored, it should make since if you look at the changes since your last review.

chrismclarke commented on .github/workflows/test-appetize.yml:
```diff
@@ -5,12 +5,37 @@ on:
+    environment: 
+      name: ${{ needs.set_deployment_name.outputs.DEPLOYMENT_NAME }}
+    env:
+      DEPLOYMENT_NAME: ${{ needs.set_deployment_name.outputs.DEPLOYMENT_NAME }}
```
nice! Yeah that makes sense, so in the first job you create the variable and then just use that output to populate to the github env here (alongside the other environment-specific variables)

chrismclarke commented on .github/workflows/test-appetize.yml:
```diff
@@ -5,12 +5,37 @@ on:
+      DEPLOYMENT_NAME: ${{ needs.set_deployment_name.outputs.DEPLOYMENT_NAME }}
     steps:
+      - name: Log Deployment/Environment Name
+        run: echo $DEPLOYMENT_NAME
```
nit(non-blocking)
I think when going through action logs it shows the environment variables anyway so no need to log beforehand, e.g.
![image](https://user-images.githubusercontent.com/10515065/228228296-a40fa63f-24b5-49e1-8893-3ce42a58af47.png)

jfmcquade replied:
Ah yeh I didn't spot that. I was thinking it could be helpful to log what environment is being used, but I note that that's also available elsewhere, at least on the PR and the specific Github Actions page
<img width="500" alt="Screenshot 2023-03-28 at 13 43 34" src="https://user-images.githubusercontent.com/64838927/228239866-8e851ec9-81f9-43bf-bdf4-0ceeaa31c69c.png">
<img width="600" alt="Screenshot 2023-03-28 at 13 43 10" src="https://user-images.githubusercontent.com/64838927/228239870-8688b07b-48ee-4550-80ef-cabaa31594cb.png">

chrismclarke reviewed (approved):
Looks great, nice to have the pattern if we want to implement similarly on any of the other actions. One minor non-blocking comment, but otherwise happy to approve and looking forward to see how it goes in production

esmeetewinkel commented:
@jfmcquade I just tested this on a PR (https://github.com/IDEMSInternational/parenting-app-ui/pull/1868) from `deployment_dev/wash` to `deployment/wash`, this is working well :) 

<img width="280" alt="image" src="https://user-images.githubusercontent.com/74557272/228844457-f9e5345e-b5bf-4d02-a26f-bb5b36d95736.png">
